gcloud/bigquery/table.py

```diff
@@ -525,3 +551,146 @@ def delete(self, client=None):
+                  of rows in the table;  and ``page_token`` is an opaque
+                  string which can be used to fetch the next batch of rows
+                  (``None`` if no further batches can be fetched).
+        """
+        client = self._require_client(client)
+        params = {}
+
+        if max_results is not None:
+            params['maxResults'] = max_results
+
+        if page_token is not None:
+            params['pageToken'] = page_token
+
+        response = client.connection.api_request(method='GET',
+                                                 path='%s/data' % self.path,
+                                                 query_params=params)
+        total_rows = response.get('totalRows')
+        page_token = response.get('pageToken')
+        rows_data = []
+
+        for row in response.get('rows', ()):
+            row_data = []
+            for field, cell in zip(self._schema, row['f']):
+                value = cell['v']
+                converter = _CELLDATA_FROM_JSON.get(field.field_type)
+                if converter is not None:
+                    value = converter(value)
+                row_data.append(value)
+            rows_data.append(tuple(row_data))
+
+        return rows_data, total_rows, page_token
+
+    def insert_all(self,
+                   rows,
+                   row_ids=None,
+                   skip_invalid_rows=None,
+                   ignore_unknown_values=None,
+                   client=None):
+        """API call:  insert table data via a POST request
+
+        See:
+        https://cloud.google.com/bigquery/reference/rest/v2/tabledata/insertAll
+
+        :type rows: list of tuples
+        :param rows: row data to be inserted
+
+        :type row_ids: list of string
+        :param row_ids: Unique ids, one per row being inserted.  If not
+                        passed, no de-duplication occurs.
+
+        :type skip_invalid_rows: boolean or ``NoneType``
+        :param skip_invalid_rows: skip rows w/ invalid data?
+
+        :type ignore_unknown_values: boolean or ``NoneType``
+        :param ignore_unknown_values: ignore columns beyond schema?
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+
+        :rtype: list of mappings
+        :returns: One mapping per row with insert errors:  the "index" key
+                  identifies the row, and the "errors" key contains a list
+                  of the mappings describing one or more problems with the
+                  row.
+        """
+        client = self._require_client(client)
+        rows_info = []
+        data = {'rows': rows_info}
+
+        for index, row in enumerate(rows):
+            row_info = {}
+
+            for field, value in zip(self._schema, row):
+                converter = _JSON_FROM_CELLDATA.get(field.field_type)
```

Comment:
With only one entry it's (effectively) not a dispatch table, it's an if statement, which we already have `if converter is not None`.